### PR TITLE
Fix duplicate entry for key campaign_rotation

### PR DESCRIPTION
**Pull request by kuzmany**
| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | "features"
| Bug fix?                               | yes
| New feature?                           | no
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | yes/no
| Related user documentation PR URL      | 
| Related developer documentation PR URL | 
| Issue(s) addressed                     | 

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Create campaign 1 - with landing page visits decision and modify contacts tags
3. Run this campaign for contact
4. Then create campain 2 - with same logic like campaign 1
5. Remove contact from campaign 1
6. Add contact to camapign 1 and campaign 2
7. Now when you come to page, you expect both campaigns are triggered
8. Without PR return error:
 
image

### CODE REVIEW

File: app/bundles/CampaignBundle/Tests/Executioner/Logger/EventLoggerTest.php [hunk 1]
```diff
@@ -74,6 +74,15 @@ public function testAllLogsAreReturnedWithFinalPersist()
         $this->assertEquals($logCollection->getValues(), $persistedLogs->getValues());
     }
 
+    public function testHydrateContactRotationsForNewLogs()
```
Comment (kuzmany) [inline app/bundles/CampaignBundle/Tests/Executioner/Logger/EventLoggerTest.php]: You should check two states of call by two separate tests methods:

setRotation() - once
leadRepository->getContactRotations() - never

and

setRotation() - once
leadRepository->getContactRotations() - once

**Comment by escopecz** [comment]
Removing a WIP PR from milestone.

**Comment by alanhartless** [comment]
@kuzmany Could you add a bit more the description please on the scenario this PR is fixing?

**Comment by kuzmany** [comment]
@alanhartless I wrote steps to reproduce. Thanks for test

**Comment by alanhartless** [comment]
Thanks!

@dongilbert fixed this for us a little differently by changing `$this->contactRotations` to be keyed by campaign then contact ID. I think either way works in the context of decisions. I think my concern with this approach is if something in the future starts using EventLogger::buildLogEntry() that will lead to fetching rotations one contact at a time. Not likely a big deal but could become a performance issue. 

@dongilbert could you  please PR your fix so @kuzmany can review and together decide which approach should be taken?

**Comment by kuzmany** [comment]
@alanhartless thanks for clarification. Wait for PR

**Comment by kuzmany** [comment]
@dongilbert @alanhartless any news?

**Comment by ShareBtech** [comment]
I have implemented this PR into a live environment where I was experiencing similar issues and it seems to have fixed the issue.  The post on the forum will more details can be found [here](https://forum.mautic.org/t/campaign-understanding-deep-diving-with-mysql-help-urgent/21550/5).

**Review by alanhartless — APPROVED** [review]
Will approve this for now as it prevents the issue. @dongilbert can provide his later as an optimization to address the scenario of a one query per contact.

**Pull request merged**